docs: lifecycle-operator non-blocking deployment functionality (#3123)
keptn/lifecycle-toolkit: docs/docs/components/lifecycle-operator/deployment-flow.md

the `KeptnApp` issues the appropriate `*Succeeded` event
and initiates the next phase.

> > **Note**
> This behavior can be changed by enabling the non-blocking
> deployment functionality by populating a
> [KeptnConfig](../../reference/crd-reference/config.md) resource and
> setting the `.spec.blockDeployment` parameter to `false`.
> More about the non-blocking deployment functionality
> can be found in the
> [Keptn non-blocking deployment section](./keptn-non-blocking.md).
> 

Reviewer: ```suggestion
!!! note
This behavior can be changed by configuring non-blocking deployments. 
More information can be found in the
[Keptn non-blocking deployment section](./keptn-non-blocking.md).

```

Reviewer: mkdocs notes may require 

<!-- markdownlint-disable MD046 >
